doc/source/release_notes.rst

```diff
@@ -1,6 +1,135 @@
 Release History
 ---------------
 
+v1.2.0 (pre-release)
+====================
+
+Features
+********
+
+* On each Signal or Device, attach a Python logger attribute named ``log``
+  with a logger name scoped by module name and the ophyd ``name`` of the
+  parent Device.
+* Signals and Devices now accept ``labels`` argument, a set of labels
+  --- presumed but not (yet) forced to be strings --- which the user can use
+  for grouping and displaying available hardware. The labels are accessible via
+  a new attribute ``_ophyd_labels_``, so name to facilitate duck-typing across
+  libraries. For example, the bluesky IPython "magics" use this to identify
+  objects for the purpose of displaying them in labeled groups.
+* Added ``tolerated_alarm`` attribute to ``EpicsMotors``, a hook to increase
+  alarm tolerance for mis-configured motors.
+* Ophyd is now fully tested to work against the experimental control layer,
+  caproto, in addition to pyepics. The control layer can also be set to 'dummy'
+  for testing without EPICS. This is configurable via the
+  ``OPHYD_CONTROL_LAYER`` environment variable.
+* Added a ``kind`` attribute to each Signal and Device, settable interactively
+  or via an argument at initiation time, which controls whether its parent
+  Device will include it in ``read()``, ``read_configuration()``, and/or
+  ``hints.fields``. This behavior was previously controlled by ``read_attrs``,
+  ``configuration_attrs``, ``_default_read_attrs``, and
+  ``_default_configuration_attrs`` on parent Devices. Those can still be used
+  for *setting* the desired state, but the source of truth is now stored
+  locally on each child Signal/Device, and
+  ``read_attrs``/``configuration_attrs`` has been re-implemented as a
+  convenience API. Documentation is forthcoming; until then we refer to you the
+  `narrative-style tests of this feature <https://github.com/NSLS-II/ophyd/blob/master/ophyd/tests/test_kind.py>`_. Also see three breaking changes, listed in a subsequent
+  section of these release notes. The existing implementation contained buggy
+  and surprising behavior, and addressing that made breaking *something*
+  unavoidable.
+* Add ``sum_all`` component to QuadEM.
+
+Bug Fixes
+*********
+
+* Allow ``DerivedSignal`` to accept a string name as its target component so
+  that it can be used inside Device, where it must defer grabbing its target to
+  initialization time.
+* Signals that start with underscores are now not renamed by ``namedtuple``.
```

Comment:
Might be worth mentioning these Components are completely omitted from the tuple.